source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -322,6 +323,7 @@ void Filter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callb
 void Filter::setEncoderFilterCallbacks(Http::StreamEncoderFilterCallbacks& callbacks) {
   Http::PassThroughFilter::setEncoderFilterCallbacks(callbacks);
   encoding_state_.setEncoderFilterCallbacks(callbacks);
+  watermark_callbacks_.setEncoderFilterCallbacks(&callbacks);
```

Comment:
ah doing this in a constructor would be hard then.  maybe some comments that the set functions must be called before the stream is used?